service/src/main/java/bio/terra/workspace/service/workspace/GcpCloudContextService.java

```diff
@@ -0,0 +1,104 @@
+package bio.terra.workspace.service.workspace;
+
+import bio.terra.workspace.db.WorkspaceDao;
+import bio.terra.workspace.service.workspace.exceptions.CloudContextRequiredException;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import bio.terra.workspace.service.workspace.model.GcpCloudContext;
+import bio.terra.workspace.service.workspace.model.Workspace;
+import java.util.Optional;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * This service provides methods for managing GCP cloud context. These methods do not perform any
+ * access control and operate directly against the {@link bio.terra.workspace.db.WorkspaceDao}
+ */
+@Component
```

Comment:
I'm a little concerned about this class causing more confusion in the future. From an outside caller perspective, it's not clear to me that `gcpCloudContextService.createGcpCloudContext` is a thin wrapper around the DAO with no auth, while `workspaceService.createGcpCloudContext` checks auth and launches a flight (and same for `deleteGcpCloudContext` in both places). This might just be an issue with the conflicting names though, and it might be clearer with a class name like `GcpCloudContextDbService` or something similar (and maybe method names like `persistGcpCloudContext`? not sure).

As (yet another) alternative, this class feels like it's pretty close to being `GcpCloudContextDao`, with the GCP cloud context methods currently in `WorkspaceDao` implemented directly here instead. That also tracks with the methods here not requiring authentication.

Reply:
Talked offline with Zach and we agreed as a way forward:
1. Rename the WorkspaceService endpoints getAuthorizedGcpContext, etc to make it the functional difference clear.
2. Merge this PR
3. Implement PF-1007 based on Will’s suggestion
4. Move the GcpContext-related endpoints from WorkspaceService to GcpCloudContextService.